src/nvim/indent.c

```diff
@@ -1292,37 +1288,33 @@ int get_lisp_indent(void)
 
           parencount = 0;
 
-          init_chartabsize_arg(&cts, curwin,
-                               (colnr_T)(that - line), amount, line, that);
           if (((*that != '"') && (*that != '\'') && (*that != '#')
                && (((uint8_t)(*that) < '0') || ((uint8_t)(*that) > '9')))) {
             int quotecount = 0;
-            while (*cts.cts_ptr
-                   && (!ascii_iswhite(*cts.cts_ptr) || quotecount || parencount)) {
-              if (*cts.cts_ptr == '"') {
+            while (*that && (!ascii_iswhite(*that) || quotecount || parencount)) {
+              if (*that == '"') {
                 quotecount = !quotecount;
               }
-              if (((*cts.cts_ptr == '(') || (*cts.cts_ptr == '[')) && !quotecount) {
+              if (((*that == '(') || (*that == '[')) && !quotecount) {
                 parencount++;
               }
-              if (((*cts.cts_ptr == ')') || (*cts.cts_ptr == ']')) && !quotecount) {
+              if (((*that == ')') || (*that == ']')) && !quotecount) {
                 parencount--;
               }
-              if ((*cts.cts_ptr == '\\') && (*(cts.cts_ptr + 1) != NUL)) {
-                cts.cts_vcol += lbr_chartabsize_adv(&cts);
+              if ((*that == '\\') && (*(that + 1) != NUL)) {
+                amount += cts_charsize(type, amount, that, (uint8_t)(*that), &arg).width;
+                that++;
               }
 
-              cts.cts_vcol += lbr_chartabsize_adv(&cts);
+              amount += cts_charsize(type, amount, that, (uint8_t)(*that), &arg).width;
+              that++;
```

Comment:
Hmm, shouldn't `utfc_next()` be used here?

Reply:
Yes, I should've used `utfc_next()` in this case